src/component/feedback/FeedbackCESContext/FeedbackCESProvider.tsx

```diff
@@ -0,0 +1,66 @@
+import React, {
+    useState,
+    ReactNode,
+    useMemo,
+    useCallback,
+    useEffect,
+} from 'react';
+import { FeedbackCES } from 'component/feedback/FeedbackCES/FeedbackCES';
+import {
+    feedbackCESContext,
+    ShowFeedbackCES,
+    IFeedbackCESContext,
+    IFeedbackCESState,
+} from 'component/feedback/FeedbackCESContext/FeedbackCESContext';
+import { useFeedbackCESSeen } from 'component/feedback/FeedbackCESContext/useFeedbackCESSeen';
+import { useFeedbackCESEnabled } from 'component/feedback/FeedbackCESContext/useFeedbackCESEnabled';
+
+interface IFeedbackProviderProps {
+    children: ReactNode;
+}
+
+export const FeedbackCESProvider = ({ children }: IFeedbackProviderProps) => {
+    const [state, setState] = useState<IFeedbackCESState>();
+    const [seen, setSeen] = useFeedbackCESSeen();
+    const enabled = useFeedbackCESEnabled();
+
+    // Store new feedback paths as seen in localStorage.
+    useEffect(() => {
+        if (state) {
+            setSeen(prev => ({
+                ...prev,
+                [state.path]: true,
+            }));
+        }
+    }, [state, setSeen]);
+
+    // Set a new feedback state iff the path is unseen and CES is enabled.
+    const showFeedbackCES: ShowFeedbackCES = useCallback(
+        value => {
+            setState(prev => {
+                const next = value instanceof Function ? value(prev) : value;
+                return !next || seen[next.path] || !enabled ? undefined : next;
+            });
+        },
+        [enabled, seen]
+    );
```

Comment:
For clarification: is everything done via localstorage? Would that be an issue for users? Are they gonna switch browsers often? I think the other in-app feedback uses a dedicated db table. Is that something we want to consider here too?

Reply:
That's much better! The seen state should now be stored on the backend.